modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj

```diff
@@ -371,6 +371,30 @@
           ;; make sure that the fake exception was thrown, and thus the query execution was retried
           (is (true? @fake-execute-called)))))))
 
+(deftest not-retry-cancellation-exception-test
+  (mt/test-driver :bigquery-cloud-sdk
+    (let [fake-execute-called (atom false)
+          orig-fn        @#'bigquery/execute-bigquery]
+      (testing "Should not retry query on cancellation"
+        (with-redefs [bigquery/execute-bigquery (fn [^BigQuery client ^String sql parameters _ _]
+                                                  ;; We dont want to retry the db setup queries, so only retry if the query contains "cancel_test"
+                                                  (if (and (re-find #"cancel_test" sql) (not @fake-execute-called))
```

Comment:
I would prefer a string more specific than this. I'm fine with appending a random UUID or prefixing with the namespace or something similar.

Reply:
I used the full test name now as the string